Added parseaddr and removed some Exception aliases that were never called

bwelboren opened the pull request:
# Description

Please include a summary of the change and which issue is fixed. List any dependencies that are required for this change.

Fixes #(issue)

## Type of change
Just put an x in the [] which are valid.
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes.
- [x] `flake8`
- [x] `python manage.py test`

# Checklist:
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings

bwelboren committed:
__has_at function returns a bool

returns a bool and uses the parseaddr function

bwelboren committed:
e is not used

bwelboren committed:
Exception alias is never used

bwelboren committed:
Exception alias never used

prabhakar267 commented on api/modules/city/views.py:
```diff
@@ -51,7 +51,7 @@ def get_city(request, city_id):
         city_visit_log = CityVisitLog(city=city, user=request.user)
         city_visit_log.save()
-    except Exception as e:
+    except Exception:
```
Please do not add files which are not relevant to the fix.
If you want to remove these unused alias, you are raise another PR

Swati4star replied:
@bjornwelboren Please remove this file from commit as there is no relevant change here.
Do the same for the other 2 files also.

prabhakar267 commented on api/modules/users/validators.py:
```diff
@@ -13,7 +15,7 @@ def _has_at(email):
     :return: boolean
     """
-    return "@" in email
+    return '@' in parseaddr(email)[1]
```
Do not modify existing validators, instead add a new one.
Also add tests for the same

bwelboren committed:
New validator function for email

Added a new validator function for validating the email with the parseaddr module as requested

bwelboren committed:
Updated validators.py

Added a test for the added function _is_email

bwelboren committed:
Update validators.py

Added _is_email to the tests

bwelboren commented:
Okay I added a new function and ran a test on it

bwelboren committed:
Reverted changes for views.py

bwelboren committed:
Delete validators.py

Swati4star commented on api/modules/currency/views.py:
```diff
@@ -32,7 +32,7 @@ def get_currency_exchange_rate(request, source_currency_code, target_currency_co
                                 result=api_response_json[query])
 
-    except Exception as e:
+    except Exception:
```
@bjornwelboren Don't delete the files. They just shouldnt be a part of this commit
